internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScannerServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.security.tools.scanmanager.webapp.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.NameValuePair;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.wso2.security.tools.scanmanager.common.external.model.Scanner;
+import org.wso2.security.tools.scanmanager.webapp.config.ScanManagerWebappConfiguration;
+import org.wso2.security.tools.scanmanager.webapp.exception.ScanManagerWebappException;
+import org.wso2.security.tools.scanmanager.webapp.util.HTTPUtil;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Scanner service implementation class.
+ */
+@Service
+public class ScannerServiceImpl implements ScannerService {
+
+    public List<Scanner> getScanners() throws ScanManagerWebappException {
+        List<Scanner> scannerList = new ArrayList<>();
+        List<NameValuePair> nameValuePairs = new ArrayList<>();
+        try {
+            ResponseEntity<String> responseEntity = HTTPUtil.sendGET(ScanManagerWebappConfiguration.getInstance()
+                    .getScannersURL("", nameValuePairs).toString(), null, null);
+            if (responseEntity != null && responseEntity.getStatusCode().is2xxSuccessful()) {
+                JSONArray responseArray = new JSONArray(responseEntity.getBody());
+                for (int arrayIndex = 0; arrayIndex < responseArray.length(); arrayIndex++) {
+                    JSONObject jsonObject = responseArray.getJSONObject(arrayIndex);
+                    ObjectMapper mapper = new ObjectMapper();
+                    Scanner scanner = mapper.readValue(jsonObject.toString(), Scanner.class);
+                    scannerList.add(scanner);
+                }
+            } else {
+                throw new ScanManagerWebappException("Unable to get the scanners");
+            }
+        } catch (IOException e) {
+            throw new ScanManagerWebappException("Unable to get the scanners", e);
+        }
+        return scannerList;
+    }
+
+    @Override
+    public Scanner getScanner(String id) throws ScanManagerWebappException {
+        List<NameValuePair> nameValuePairs = new ArrayList<>();
+        Scanner scanner = null;
+        try {
+            ResponseEntity<String> responseEntity = HTTPUtil.sendGET(ScanManagerWebappConfiguration.getInstance()
+                    .getScannersURL(id, nameValuePairs).toString(), null, null);
+            if (responseEntity != null && responseEntity.getStatusCode().is2xxSuccessful()) {
+                JSONObject jsonObject = new JSONObject(responseEntity.getBody());
+                ObjectMapper mapper = new ObjectMapper();
+                scanner = mapper.readValue(jsonObject.toString(), Scanner.class);
```

Comment:
In this situation, it is better to return from here and avoid creating the null instance (`Scanner scanner = null;`).